Library/docs/source/learn/overview/mcs.md

```diff
@@ -0,0 +1,334 @@
+
+### Result Objects
+
+A Result Object contains response data retrieved from an asynchronous operation in CSP. The object transforms raw data received from the services into interoperable, strongly typed formats.
+
+An example of a Result Object is AssetCollectionResult.The AssetCollectionResult class is a result object that stores information about an asset collection.
+
+```
+class CSP_API AssetCollectionResult : public csp::systems::ResultBase
+{
+public:
+	AssetCollection& GetAssetCollection();
+	const AssetCollection& GetAssetCollection() const;
+private:
+	AssetCollectionResult(void*);
+	void OnResponse(const csp::services::ApiResponseBase* ApiResponse) override;
+	AssetCollection AssetCollection;
+};
+```
+
+* The Result Object contains an associated object, AssetCollection, stored as a private member.
+
+* The AssetCollectionResult defines two getters to get the AssetCollection by reference or by const reference.
+
+* The `AssetCollectionResult` includes an `OnResponse` method that is used internally within CSP to transform the DTO (Data Transfer Object)  received from services into a strongly typed CSP object, `AssetCollection`.
+
+## Interaction Between Callbacks and Result Objects
+
+For asynchronous operations, the interaction between callbacks and result objects is essential. 
+
+Here is an example of a workflow:
+
+**1\. Client Code Initiates Request**:
+
+```
+csp::systems::AssetCollectionResultCallback GetAssetCollectionByIdCallback = [Callback, this](const csp::systems::AssetCollectionResult& Result)
+{
+	if (Response == EOKOResponse::Success)
+	{
+		const csp::systems::AssetCollection& RetrievedAssetCollection = Result.GetAssetCollection();
+		// Process the retrieved asset collection...
+	}
+};
+
+AssetSystem->GetAssetCollectionById(MyAssetCollectionId, GetAssetCollectionByIdCallback);
+```
+
+**2\. Request Handling**:
+
+1. The client code defines a callback (`GetAssetCollectionByIdCallback`) as a lambda expression.
+
+2. The client code calls the GetAssetCollectionById method, passing the asset collection ID and the callback as parameters.  
+   
+
+**3\. Services Response**:
+
+1. The services process the request and send a response.
+
+2. The CSP ResponseHandler generates an AssetCollectionResult object from the HTTP response.  
+   
+
+**4\. Callback Execution**:
+
+1. Within CSP, The OnResponse method of AssetCollectionResult parses the DTO into a CSP object.
+
+2. The callback is executed, allowing the client application to use the parsed data via the result object's getters.
+
+## Mapping Magnopus Cloud Services to CSP Systems
+
+Mapping MCS to CSP systems is crucial to understanding how various services and functionalities are used within CSP. This process involves aligning the core components of MCS with the corresponding systems and subsystems in CSP to ensure they work together smoothly and effectively to establish a strong operational connection
```

Comment:
I _think_ this should be "to understand" (not "to understanding"), however this might be an oddity in English grammar that I do not know of, where this might be correct

Reply:
Fixed in latest. Thanks!